elastic/elasticsearch
# Support zone ID in IPv6 addresses. Issue #37107

Issue https://github.com/elastic/elasticsearch/issues/37107 describes problems with IPv6 addresses that contain a zone ID.

The scope of this PR is to ignore the zone ID if any.
This on itself will address the reported bug.

Keep in mind this is the first PR of a series.  Subsequent PR will properly read the zone ID and construct IPv6 addresses correctly.

Here's a summary of the code changes 
- change ipStringToBytes to ignore anything after a percent
- make sure percent does not appear before dot
- make sure percent does not appear before colon
- can only contain one percent symbol
- make sure mixed IPv6IPv4 addresses support zoneId
- test an address with dots and colons and percent
- test an address with dots and percents
- test an address with colons and percents
- make sure percent does not appear right after dot
- make sure percent does not appear right after colon
